packages/grid_client/src/clients/tf-grid/contracts.ts

```diff
@@ -301,13 +326,216 @@ class TFContracts extends Contracts {
     });
   }
 
+  /** @deprecated */
   async contractLock(options: ContractLockOptions) {
     const res = await super.contractLock(options);
     const amountLocked = new Decimal(res.amountLocked);
     res.amountLocked = amountLocked.div(10 ** 7).toNumber();
     return res;
   }
 
+  /**
+   * Function to convert USD to TFT
+   * @param {Decimal} USD the amount in USD.
+   * @returns  {Decimal} The amount in TFT.
+   */
+  private async convertToTFT(USD: Decimal) {
+    try {
+      const tftPrice = (await this.client.tftPrice.get()) ?? 0;
+      const tft = new currency(tftPrice).convertUSDtoTFT({ amount: USD.toNumber() });
+      return new Decimal(tft);
+    } catch (error) {
+      throw new GridClientError(`Failed to convert to mTFT due: ${error}`);
+    }
+  }
+  /**
+   * list all contracts, this is not restricted with the items counts
+   * this basically check if the count is larger than the page size, it make another request with the item count as the size pram.
+   * @param {GridProxyClient} proxy will be used to list the contracts
+   * @param {Partial<ContractsQuery>} queries
+   * @returns
+   */
+  async listAllContracts(proxy: GridProxyClient, queries: Partial<ContractsQuery>) {
+    const contracts = await proxy.contracts.list({ ...queries, retCount: true });
+    if (contracts.data.length < contracts.count!) {
+      return (
+        await proxy.contracts.list({
+          ...queries,
+          size: contracts.count!,
+        })
+      ).data;
+    } else return contracts.data;
+  }
+
+  /**
+   * Calculate total IPV4 and total overdraft on all node contracts on rented node.
+   * @description In node contracts on rented nodes we need to have both the overdraft to be added to rent contract overdraft.
+   * please note that the unbilled NU amount is added to the total overdraft exactly as the in the other contract types.
+   * the IPV4 cost to add to the estimated cost of the rent contract.
+   */
+  private async getContractsCostOnRentedNode(nodeId: number, proxy: GridProxyClient) {
```

Comment:
This function calculates two separate costs:
1. The overdue amount for all node contracts associated with a rent contract (total overdraft + unbilled NU cost).
2. The IP cost for all node contracts per month.

In my opinion, it would be better to handle these calculations with separate functions.

Reply:
Actually, I put both of them in one place to reduce listing the contracts again, here already have the contracts and their count, i was thinking about move the ip calculations to separate function and call it here but this will change nothing imo, what do you think?

Reply:
This is because this function does a lot of things.
What about splitting it into three separate functions?
The three new functions would be:
1. `getNodeContractsOnRentedNode`
2. `calculateNodeContractsOverdue`
3. `calculateIPCostPerMonth`
This way, you can retrieve the contracts from the first function and then pass them to the second and third functions.